packages/utils/telemetry-utils/src/error.ts

```diff
@@ -21,6 +21,25 @@ import {
 } from "./errorLogging";
 import { IFluidErrorBase } from "./fluidErrorBase";
 
+/**
+ * Throws a UsageError with the given message if the condition is not met.
+ * Use this API when `false` indicates a precondition is not met on a public API (for any FF layer).
+ *
+ * @param condition - The condition that should be true, if the condition is false a UsageError will be thrown.
+ * @param message - The message to include in the error when the condition does not hold.
+ * @param props - Telemetry props to include on the error when the condition does not hold.
+ * @internal
+ */
+export function validatePrecondition(
```

Comment:
Should this go in `core-utils`, since `ITelemetryBaseProperties` is in `core-interfaces`, and this isn't really telemetry-related?

Reply:
Yeah, I started there but moved it here out of necessity.  We'll need to move the error classes (I want to use UsageError) and they extend LoggingError.  I think it's fine for LoggingError to move to core-utils.  In the meantime, this can sit here, but long term I see this moving alongside `assert` and getting assert code support, and updating `assert` to emit a `LoggingError` as well.